File: 30 days of code JAVA/day26.java


import java.util.Scanner;

public class Solution {

    public static void main(String[] args) {
        /* Enter your code here. Read input from STDIN. Print output to STDOUT. Your class should be named Solution. */

        Scanner sc = new Scanner(System.in);

        int actualDay = sc.nextInt();
        int actualMonth = sc.nextInt();
        int actualYear = sc.nextInt();

        int expectedDay = sc.nextInt();
        int expectedMonth = sc.nextInt();
        int expectedYear = sc.nextInt();


        int fine = computeFine(actualDay, actualMonth, actualYear, expectedDay, expectedMonth, expectedYear);

        System.out.println(fine);
    }

    static int computeFine(int actualDay, int actualMonth, int actualYear, int expectedDay, int expectedMonth, int expectedYear) {

        int fine = 0;

        if (actualYear > expectedYear) {
            fine = 10000;
        }
        else if (actualYear == expectedYear) {

            if (actualMonth > expectedMonth) {
                fine = (actualMonth - expectedMonth) * 500;
            }
            else if (actualMonth == expectedMonth) {

                if (actualDay > expectedDay) {
                    fine = (actualDay - expectedDay) * 15;
                }
            }
        }

        return fine;
    }
}

File: 30 days of code( python)/Day 26.py
# Enter your code here. Read input from STDIN. Print output to STDOUT

returned_date = raw_input().split(' ')
due_date = raw_input().split(' ')

def fine_calc(returned_date, due_date):
    if (str(''.join(returned_date)) == str(''.join(due_date))):
        return 0
    
    
    
    returned_date = [int(t) for t in returned_date]
    due_date = [int(t) for t in due_date]

    
    if ((returned_date[2] == due_date[2]) and (returned_date[1] == due_date[1]) and (returned_date[0] > due_date[0])):
        # Case 2
        return (15 * (returned_date[0] - due_date[0]))
    
    if ((returned_date[1] > due_date[1]) and (returned_date[2] == due_date[2])):
        # Case 3
        return (500 * (returned_date[1] - due_date[1]))
        
        
    if (returned_date[2] > due_date[2]):
        # Case 4
        return 10000
    
    if(returned_date[2] < due_date[2]):
        # Early (different year)
        return 0
    
    if (returned_date[2] == due_date[2]) and (returned_date[1] < due_date[1]):
        # Early (same year, different month)
        return 0
    
    return 0
        
ans = fine_calc(returned_date, due_date)    
print ans
